### ISSUE TITLE
crash on bad default argument

### ISSUE THREAD

**Issue by ec04fc15-fa35-46f2-80e1-5d271f2ef708** [llvm/llvm-project#21336]
|  |  |
| --- | --- |
| Bugzilla Link | [20962](https://llvm.org/bz20962) |
| Version | trunk |
| OS | Linux |
| CC | @DougGregor |

## Extended Description 
This causes clang to crash:

```cpp
template<typename T> struct S { void f(int k = g(k)()); };
void x() { S<int>().f(); }
```

The interesting part of the backtrace:

7  clang-3.5       0x0000000003645f04 clang::Sema::CheckImplicitConversions(clang::Expr*, clang::SourceLocation) + 132
8  clang-3.5       0x0000000003646812 clang::Sema::CheckCompletedExpr(clang::Expr*, clang::SourceLocation, bool) + 66
9  clang-3.5       0x000000000382b215 clang::Sema::BuildCXXDefaultArgExpr(clang::SourceLocation, clang::FunctionDecl*, clang::ParmVarDecl*) + 1253

We're presumably calling CheckCompletedExpr on a null expression.

**Comment by shafik** [comment]
This no longer reproduces on trunk: https://godbolt.org/z/xbs9WMzxv

**Comment by Endilll** [comment]
Fixed in Clang 3.7: https://godbolt.org/z/z48sMhE4j
